ownCloud/Client/ClientQueryViewController.swift

```diff
@@ -849,3 +935,48 @@ extension ClientQueryViewController: UITableViewDragDelegate {
 	}
 
 }
+
+// MARK: - UIImagePickerControllerDelegate
+extension ClientQueryViewController: UIImagePickerControllerDelegate {
+
+	func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
+
+		var name: String = "test.jpg"
+		var url: URL = URL(string: "https://demo.owncloud.org")!
+
+		if let imageURL = info[UIImagePickerControllerImageURL] as? URL {
+			name = imageURL.lastPathComponent
+			url = imageURL
+		}
+
+		if let movieURL = info[UIImagePickerControllerMediaURL] as? URL {
+			name = movieURL.lastPathComponent
+			url = movieURL
+		}
+
+		if let imageAsset = info[UIImagePickerControllerPHAsset] as? PHAsset {
+			let resources = PHAssetResource.assetResources(for: imageAsset)
+			name = resources[0].originalFilename
+		}
+
+		upload(itemURL: url, name: name)
```

Comment:
… and `upload(itemURL:name:)` only be called if `url` is non-nil.

Reply:
Done